test/kubemark/cloudprovider.go

```diff
@@ -0,0 +1,342 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubemark
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/kubernetes/pkg/api"
+	kube_client "k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	informers "k8s.io/kubernetes/pkg/client/informers/informers_generated/externalversions"
+
+	"github.com/golang/glog"
+)
+
+const (
+	namespaceKubemark = "kubemark"
+	hollowNodeName    = "hollow-node"
+	nodeGroupLabel    = "autoscaling.k8s.io/nodegroup"
+)
+
+// Provider is a simplified version of cloud provider for kubemark. It allows
+// to add and delete nodes from a kubemark cluster and introduces nodegroups
+// by applying labels to the kubemark's hollow-nodes.
+type Provider struct {
+	nodeTemplate    *apiv1.ReplicationController
+	externalCluster ExternalCluster
+	kubemarkCluster KubemarkCluster
+}
+
+// ExternalCluster is used to communicate with the external cluster that hosts
+// kubemark, in order to be able to list, create and delete hollow nodes
+// by manipulating the replication controllers.
+type ExternalCluster struct {
+	rcLister  cache.SharedInformer
+	podLister cache.SharedInformer
+	client    kube_client.Interface
+}
+
+// KubemarkCluster is used to delete nodes from kubemark cluster once their
+// respective replication controllers have been deleted and the nodes have
+// become unready. This is to cover for the fact that there is no proper cloud
+// provider for kubemark that would care for deleting the nodes.
+type KubemarkCluster struct {
+	client            kube_client.Interface
+	nodeLister        cache.SharedInformer
+	nodesToDelete     map[string]bool
+	nodesToDeleteLock sync.Mutex
+}
+
+// NewProvider creates Provider using the privided clients to talk to external
+// and kubemark clusters. It returns an error if it fails to sync data from
+// any of the two clusters.
+func NewProvider(externalClient kube_client.Interface, kubemarkClient kube_client.Interface, stop <-chan struct{}) (*Provider, error) {
+	externalInformerFactory := informers.NewSharedInformerFactory(externalClient, 0)
+	kubemarkInformerFactory := informers.NewSharedInformerFactory(kubemarkClient, 0)
+	manager := &Provider{
+		externalCluster: ExternalCluster{
+			rcLister:  externalInformerFactory.InformerFor(&apiv1.ReplicationController{}, newReplicationControllerInformer),
+			podLister: externalInformerFactory.InformerFor(&apiv1.Pod{}, newPodInformer),
+			client:    externalClient,
+		},
+		kubemarkCluster: KubemarkCluster{
+			nodeLister:        kubemarkInformerFactory.InformerFor(&apiv1.Node{}, newNodeInformer),
```

Comment:
Since you're getting a normal informer from the factory (unlike the 2 informers above which are namespaced), I guess just doing a `kubemarkInformerFactory.Core().V1().Nodes()` here should be enough.